packages/Telephony/Common/RegexAggregatorInput.cs

```diff
@@ -0,0 +1,114 @@
+        public StringExpression Property { get; set; }
+
+        /// <summary>
+        /// Gets or sets the activity to send to the user.
+        /// </summary>
+        /// <value>
+        /// An activity template.
+        /// </value>
+        [JsonProperty("prompt")]
+        public ITemplate<Activity> Prompt { get; set; }
+
+        /// <inheritdoc/>
+        public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            ITemplate<Activity> template = this.Prompt ?? throw new InvalidOperationException($"InputDialog is missing Prompt.");
+            IMessageActivity msg = await this.Prompt.BindAsync(dc, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            if (msg != null && string.IsNullOrEmpty(msg.InputHint))
+            {
+                msg.InputHint = InputHints.ExpectingInput;
+            }
+
+            var properties = new Dictionary<string, string>()
+                {
+                    { "template", JsonConvert.SerializeObject(template) },
+                    { "result", msg == null ? string.Empty : JsonConvert.SerializeObject(msg, new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore }) },
+                };
+            TelemetryClient.TrackEvent("GeneratorResult", properties);
+
+            await dc.Context.SendActivityAsync(msg, cancellationToken).ConfigureAwait(false);
+
+            return new DialogTurnResult(DialogTurnStatus.Waiting);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken = default)
+        {
+            //Get value of temrination string from expression
+            string regexPattern = this.TerminationConditionRegexPattern?.GetValue(dc.State);
+
+            //append the message to the aggregation memory state
+            var existingAggregation = dc.State.GetValue(AggregationDialogMemory, () => string.Empty);
+            existingAggregation += dc.Context.Activity.Text;
+
+            //Is the current aggregated message the termination string?
+            if (Regex.Match(existingAggregation, regexPattern).Success)
+            {
+                //If so, save it to the output property and end the dialog
+                //Get property from expression
+                string property = this.Property?.GetValue(dc.State);
+
+                dc.State.SetValue(property, existingAggregation);
+                return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                //else, save the updated aggregation and end the turn
+                dc.State.SetValue(AggregationDialogMemory, existingAggregation);
+                return new DialogTurnResult(DialogTurnStatus.Waiting);
+            }
```

Comment:
not blocking: do we need another condition if x length has reached (or in future silence detection here), just trying to understand when it will stop

Reply:
+1, seems risky to always return waiting unless the regex is met and not allow interruptions. Given that this is a modal dialog, should we have an add interruptions control like we have in CommandDialog?

@goldenwitch an interesting thing I'd be interested to see in a test, like we have for CommandDialog, is test that dialogs are interruptable. So for example what happens if I press to keys and then say help?

Reply:
As a bot developer I should be able to at least allow interruptions, and we should discuss whether as a bot developer I should be able to control whether my dialog allows or not interruptions. Thoughts?

Reply:
Yeah definitely need interruptions to be supported. I need to sit down and validate all of those scenarios with tests.